README.md

```diff
@@ -1,2 +1,91 @@
 # queries
+
 Hosts reusable log queries which are built into a single queries.json file.
+
+## Query database structure
+
+```yaml
+queries:
+  pip-no-matching-distribution:  # also known as query "id", (sort key)
```

Comment:
I don't see much sense in those ids, maybe better to have just numbers.

Reply:
Numbers do not have any meaning and they do not help with sorting. The reason why I wanted numeric id is to enforce pattern-creator to pick meaningful names, names that also help sorting problems from the same area one after another. Example all pip errors are supposed to start with pip.

That is helpful in two areas:
- when new pattern is added, it will assure it is added between similar patterns, so the chance of creating overlapping or duplicating patterns if far lower.
- when we display matches, we sort them based on their id, making it easier to show them.

I uses as inspiration pylint violations. The tool now promotes use of human identifications instead of numeric ones. Take a look at https://github.com/janjur/readable-pylint-messages and you will understand why text is better than number. I even started to add human ids to ansible-lint, for the same reasons. Is very hard to guess what `E123` points to, but `avoid-foo-var` is more human-friendly.

Keep in mind that in last revision, name is no longer required, id is used when name is not defined.